src/libraries/System.Globalization/tests/CompareInfo/CompareInfoTests.IsPrefix.cs

```diff
@@ -17,81 +17,96 @@ public class CompareInfoIsPrefixTests
+            yield return new object[] { s_invariantCompare, "FooBA\u0300R", "FooB\u00C0R", CompareOptions.IgnoreNonSpace, true, 7 };
+            yield return new object[] { s_invariantCompare, "o\u0308", "o", CompareOptions.None, false, 0 };
+            yield return new object[] { s_invariantCompare, "o\u0308", "o", CompareOptions.Ordinal, true, 1 };
+            yield return new object[] { s_invariantCompare, "o\u0000\u0308", "o", CompareOptions.None, true, 1 };
 
             // Weightless comparisons
-            yield return new object[] { s_invariantCompare, "", "\u200d", CompareOptions.None, true };
+            yield return new object[] { s_invariantCompare, "", "\u200d", CompareOptions.None, true, 0 };
+            yield return new object[] { s_invariantCompare, "\u200dxy", "x", CompareOptions.None, true, 2 };
 
             // Surrogates
-            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800\uDC00", CompareOptions.None, true };
-            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800\uDC00", CompareOptions.IgnoreCase, true };
-            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800", CompareOptions.Ordinal, true };
-            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800", CompareOptions.OrdinalIgnoreCase, true };
+            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800\uDC00", CompareOptions.None, true, 2 };
+            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800\uDC00", CompareOptions.IgnoreCase, true, 2 };
+            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800", CompareOptions.Ordinal, true, 1 };
+            yield return new object[] { s_invariantCompare, "\uD800\uDC00", "\uD800", CompareOptions.OrdinalIgnoreCase, true, 1 };
 
             // Malformed Unicode - Invalid Surrogates (there is nothing special about them, they don't have a special treatment)
-            yield return new object[] { s_invariantCompare, "\uD800\uD800", "\uD800", CompareOptions.None, true };
-            yield return new object[] { s_invariantCompare, "\uD800\uD800", "\uD800\uD800", CompareOptions.None, true };
+            yield return new object[] { s_invariantCompare, "\uD800\uD800", "\uD800", CompareOptions.None, true, 1 };
+            yield return new object[] { s_invariantCompare, "\uD800\uD800", "\uD800\uD800", CompareOptions.None, true, 2 };
 
             // Ignore symbols
-            yield return new object[] { s_invariantCompare, "Test's can be interesting", "Tests", CompareOptions.IgnoreSymbols, true };
-            yield return new object[] { s_invariantCompare, "Test's can be interesting", "Tests", CompareOptions.None, false };
+            yield return new object[] { s_invariantCompare, "Test's can be interesting", "Tests", CompareOptions.IgnoreSymbols, true, 6 };
```

Comment:
Good call on adding more tests. This uncovered a latent bug, which I filed at https://github.com/dotnet/runtime/issues/40803. Not addressing that bug as part of this PR because I don't think we have the cycles to properly investigate it right now.